stdlib/sys/file.gr

```diff
@@ -248,7 +248,12 @@ let rec combineRightsHelp = (acc, dirflags) => {
         PathUnlinkFile => _RIGHT_PATH_UNLINK_FILE,
         PollFdReadwrite => _RIGHT_POLL_FD_READWRITE,
         SockShutdown => _RIGHT_SOCK_SHUTDOWN,
-        _ => fail "Unknown file right",
+        _ => {
+          ignore(fail "Unknown file right")
```

Comment:
Just to mute the warning about it being an unsound type since it's not used!

Reply:
Sorry, I'm still confused. Why does the `a` return value of `fail "Unknown file right"` not unify to `WasmI64` (in which case it wouldn't need to be ignored)?

Reply:
It does! But at the WebAssembly level, Grain "any" types are i32s but this block needs to return an i64 like the others. We got around this before by this being a `call_indirect`—we just lied about what type `fail` returned. At runtime, instead of raising `Failure`, this is a call_indirect typechecking error.

Essentially, we need a compiler primitive for `fail` which can print the error and then generate `unreachable`. (But we can't do that until the enum refactor is done because `Failure` would need to be baked into the compiler.)

Reply:
(And now since this is a regular `call` instruction, we can't lie about the return type.)

Reply:
Ah, okay, that makes sense. Now I understand why it was needed as part of this PR. Thanks 👍

Reply:
@ospencer Do we have an issue for your comment above? If not, please create it as a follow-up to baking Enum into the compiler.

Reply:
https://github.com/grain-lang/grain/issues/1192